Neos.ContentRepository.Core/Classes/Infrastructure/DbalSchemaFactory.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\Infrastructure;
+
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Types\Types;
+use Neos\ContentRepository\Core\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Core\NodeType\NodeTypeName;
+use Neos\ContentRepository\Core\SharedModel\Node\NodeAggregateId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+
+/**
+ * Provide doctrine DBAL column schema definitions for common types in the content repository to
+ * produce consistent columns across projections.
+ *
+ * @internal Because we might need to check for platform later on and generally change the input and output format of functions within.
+ */
+final class DbalSchemaFactory
+{
+    /**
+     * The NodeAggregateId is limited to 64 ascii characters and therefore we should do the same in the database.
+     *
+     * @see NodeAggregateId
+     */
+    public static function addColumnForNodeAggregateId(Table $table, string $columnName, bool $notNull): Table
+    {
+        $table->addColumn($columnName, Types::STRING)
+                ->setLength(64)
+                ->setNotnull($notNull)
+                ->setCustomSchemaOption('charset', 'ascii')
+                ->setCustomSchemaOption('collation', 'ascii_general_ci');
+
+        return $table;
+    }
+
+    /**
+     * The ContentStreamId is generally a UUID, therefore not a real "string" but at the moment a strified identifier,
+     * we can safely store it as binary string as the charset doesn't matter
+     * for the characters we use (they are all asscii).
+     *
+     * We should however reduce the allowed size to 36 like suggested
+     * here in the schema and as further improvement store the UUID in a more DB friendly format.
+     *
+     * @see ContentStreamId
+     */
+    public static function addColumnForContentStreamId(Table $table, string $columnName, bool $notNull): Table
+    {
+        $table->addColumn($columnName, Types::STRING)
+            ->setLength(36)
+            ->setNotnull($notNull)
```

Comment:
If we decide against changing it as suggested above, I would at least suggest not to copy the double negative for nullable columns
```suggestion
    public static function addColumnForContentStreamId(Table $table, string $columnName, bool $nullable): Table
    {
        $table->addColumn($columnName, Types::STRING)
            ->setLength(36)
            ->setNotnull(!$nullable)
```

Reply:
wanted to keep it as close as possible to the original syntax so as to not mess things up, doing this you need to carefully inverse everything that was there before :D But probably a good call. 

Right now I have no other task so I can adjust the syntax as well, but I assime there would be more important stuff or either of us to spend time on? After all this is all internal and could be refactored later.